### PR TITLE
hwmv2: port STM32F1 SoCs and boards

### PR DESCRIPTION
**Pull request by erwango**

### CODE REVIEW

File: soc/v2/st_stm32/common/Kconfig.defconfig.series [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+# Copyright (c) 2017, I-SENSE group of ICCS
+# SPDX-License-Identifier: Apache-2.0
+
+# Here are set all the Kconfig symbols common to the whole STM32 family
+
+if SOC_FAMILY_STM32
+
+config CORTEX_M_SYSTICK
+	default n if STM32_LPTIM_TIMER
```
Comment (nordicjm) [inline soc/v2/st_stm32/common/Kconfig.defconfig.series:11]: This file looks to have creeped back in again (duplicate)

File: boards/v2/st/stm3210c_eval/stm3210c_eval_defconfig [hunk 2]
```diff
@@ -1,11 +1,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # Zephyr Kernel Configuration
-CONFIG_SOC_SERIES_STM32F1X=y
-
 # Platform Configuration
```
Comment (nordicjm) [inline boards/v2/st/stm3210c_eval/stm3210c_eval_defconfig:4]: Headings can go

File: boards/v2/others/stm32_min_dev/board.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+board:
+  name: stm32_min_dev
```
Comment (nordicjm) [inline boards/v2/others/stm32_min_dev/board.yml:2]: this board has not had the _defconfig files updated
```
CONFIG_SOC_SERIES_STM32F1X=y
CONFIG_SOC_STM32F103X8=y
```
And the Kconfig.defconfig file can be deleted, Kconfig.board should be split and renamed with selection disabled

File: soc/v2/st_stm32/common/Kconfig.defconfig.series (hunk 1 above)
Comment (erwango) [reply soc/v2/st_stm32/common/Kconfig.defconfig.series]: Seen it as well. This is fixed now.

File: boards/v2/others/stm32_min_dev/board.yml (hunk 3 above)
Comment (erwango) [reply boards/v2/others/stm32_min_dev/board.yml:2]: Yeah, I fully reviewed this board variant definition scheme

**Review by tejlmand — APPROVED** [review]

**Review by sidcha — APPROVED** [review]

**Pull request merged**